ast/ast.go

```diff
@@ -362,11 +364,31 @@ func (e *MemberExpression) Copy() Node {
 	*ne = *e
 
 	ne.Object = e.Object.Copy().(Expression)
-	ne.Property = e.Property.Copy().(Expression)
+	ne.Property = e.Property.Copy().(*Identifier)
 
 	return ne
 }
 
+// IndexExpression represents indexing into an array
+type IndexExpression struct {
+	BaseNode
+	Object Expression `json:"object"`
```

Comment:
Is there a reason we still have `IndexExpression`? Is there any difference between it and `MemberExpression`?

Reply:
In the AST a bracketed expression such as `r["name"]` or `r[0]` is considered an index expression. A dot signifies a member expression.

Reply:
@jsternberg Also the type inference needs to know the difference. Javascript didn't need to differentiate them since types are checked at runtime (We modeled the ast off the Javascript ast). With Flux we need to be able to infer the types of the expressions so the difference is important.